rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -112,17 +119,97 @@ def run_check(cls, results_dir, **kwargs):
+            output = os.path.join(cls.results_dir, old_name + '.txt')
             try:
                 ret_code = ProcessHelper.run_subprocess(command, output_file=output)
             except OSError:
                 raise CheckerNotFoundError("Checker '%s' was not found or installed." % cls.CMD)
 
             if int(ret_code) & settings.ABIDIFF_ERROR and int(ret_code) & settings.ABIDIFF_USAGE_ERROR:
                 raise RebaseHelperError('Execution of %s failed.\nCommand line is: %s' % (cls.CMD, cmd))
-            if int(ret_code) == 0:
-                text = 'ABI of the compared binaries in package %s are equal.' % old_name
-            else:
-                text = 'ABI of the compared binaries in package %s are not equal.' % old_name
-            reports[output] = text
-        return reports
+            reports[old_name] = int(ret_code)
+        return {'packages': cls.parse_abi_logs(reports), 'abi_changes': cls.abi_changes, 'path': cls.results_dir}
+
+    @classmethod
+    def parse_abi_logs(cls, reports):
+        """
+        Parse abipkgdiff logs
+        :param reports: dictionary with paths to the logs of the produced (sub)packages
+        :return: returns packages names with its abipkgdiff changes
+        """
+        def parse_changes(lines):
+            """
+            Parses each line with abipkgdiff output info
+            :param lines: lines of changes
+            :return: dictionary of list of changes
+
+            Example abipkgdiff line:
+            '  Functions changes summary: 3 Removed, 0 Changed, 0 Added functions (4 filtered out)'
+            """
+            summary_re = re.compile(r'''^
+            \s+(?P<kind>[\w\s]+changes\s+summary):\s+
+            (?P<changes>.+)
+            $
+            ''', re.VERBOSE)
+            changes_re = re.compile(r'''
+            (?P<count>\d+)\s+
+            (?P<what>Added|Changed|Removed)(\s+functions)?
+            (\s+\((?P<filtered_out>\d+)\s+filtered\s+out\))?
+            ''', re.VERBOSE)
+            result = {}
+            for line in lines:
+                ms = summary_re.match(line)
+                if ms:
+                    ds = ms.groupdict()
+                    result[ds['kind']] = {}
+                    for mc in changes_re.finditer(ds['changes']):
+                        dc = mc.groupdict()
+                        if int(dc['count']) or dc['filtered_out']:
+                            result[ds['kind']][dc['what']] = {'count': dc['count'], 'filtered_out': dc['filtered_out']}
+            return result
+
+        pkgs = {}
+        for pkg, ret_code in six.iteritems(reports):
+            cur_pkg = {}
+            if ret_code:
+                with open(os.path.join(cls.results_dir, pkg + '.txt'), 'r') as f:
+                    cur_pkg = parse_changes(f.readlines())
+                    cls.abi_changes = True
+            # If no abi changes for the package, store empty dictionary
```

Comment:
Just a nitpick, but this comment would probably make more sense if it was moved 5 lines up.